src/librustc_back/svh.rs

```diff
@@ -100,25 +107,23 @@ impl Svh {
             attr.node.value.hash(&mut state);
         }
 
-        let hash = state.finish();
-        return Svh {
-            hash: (0..64).step_by(4).map(|i| hex(hash >> i)).collect()
-        };
-
-        fn hex(b: u64) -> char {
-            let b = (b & 0xf) as u8;
-            let b = match b {
-                0 ... 9 => '0' as u8 + b,
-                _ => 'a' as u8 + b - 10,
-            };
-            b as char
+        Svh {
+            raw: state.finish(),
         }
     }
 }
 
+impl Hash for Svh {
+    fn hash<H>(&self, state: &mut H) where H: Hasher {
+        // We have to hash a &str since that's what the old implementations did, and otherwise we
+        // break the abi
+        &self.to_string()[..].hash(state);
```

Comment:
It's actually possible this is not necessary any more, although I believe that the u64 and the String will hash differently. I believe this was a first guess at what was happening before when I hit the endianness bug above (My stage1 would refuse to compile anything that depended on libc or core because it believed they were out of date)